src/meshes/p4est_mesh_view.jl

```diff
@@ -0,0 +1,192 @@
+# Then, within Trixi2Vtk, the P4estMeshView and its node coordinates are reconstructured from
+# these attributes for plotting purposes
+function save_mesh_file(mesh::P4estMeshView, output_directory, timestep,
+                        mpi_parallel::False)
+    # Create output directory (if it does not exist)
+    mkpath(output_directory)
+
+    # Determine file name based on existence of meaningful time step
+    if timestep > 0
+        filename = joinpath(output_directory, @sprintf("mesh_%09d.h5", timestep))
+        p4est_filename = @sprintf("p4est_data_%09d", timestep)
+    else
+        filename = joinpath(output_directory, "mesh.h5")
+        p4est_filename = "p4est_data"
+    end
+
+    p4est_file = joinpath(output_directory, p4est_filename)
+
+    # Save the complete connectivity and `p4est` data to disk.
+    save_p4est!(p4est_file, mesh.parent.p4est)
+
+    # Open file (clobber existing content)
+    h5open(filename, "w") do file
+        # Add context information as attributes
+        attributes(file)["mesh_type"] = get_name(mesh)
+        attributes(file)["ndims"] = ndims(mesh)
+        attributes(file)["p4est_file"] = p4est_filename
+        attributes(file)["cell_ids"] = mesh.cell_ids
+
+        file["tree_node_coordinates"] = mesh.parent.tree_node_coordinates
+        file["nodes"] = Vector(mesh.parent.nodes) # the mesh uses `SVector`s for the nodes
+        # to increase the runtime performance
+        # but HDF5 can only handle plain arrays
+        file["boundary_names"] = mesh.parent.boundary_names .|> String
+    end
+
+    return filename
+end
+
+# Interpolate tree_node_coordinates to each quadrant at the specified nodes
+function calc_node_coordinates!(node_coordinates,
+                                mesh::P4estMeshView{2, NDIMS_AMBIENT},
+                                nodes::AbstractVector) where {NDIMS_AMBIENT}
+    # We use `StrideArray`s here since these buffers are used in performance-critical
+    # places and the additional information passed to the compiler makes them faster
+    # than native `Array`s.
+    tmp1 = StrideArray(undef, real(mesh),
+                       StaticInt(NDIMS_AMBIENT), static_length(nodes),
+                       static_length(mesh.parent.nodes))
+    matrix1 = StrideArray(undef, real(mesh),
+                          static_length(nodes), static_length(mesh.parent.nodes))
+    matrix2 = similar(matrix1)
+    baryweights_in = barycentric_weights(mesh.parent.nodes)
+
+    # Macros from `p4est`
+    p4est_root_len = 1 << P4EST_MAXLEVEL
+    p4est_quadrant_len(l) = 1 << (P4EST_MAXLEVEL - l)
+
+    trees = unsafe_wrap_sc(p4est_tree_t, mesh.parent.p4est.trees)
+
+    for tree_view in eachindex(mesh.cell_ids)
+        tree = mesh.cell_ids[tree_view]
+        offset = trees[tree_view].quadrants_offset
+        quadrants = unsafe_wrap_sc(p4est_quadrant_t, trees[tree].quadrants)
+
+        for i in eachindex(quadrants)
+            element = offset + i
```

Comment:
Have you tested this for `initial_refinement_level` greater than zero? It seems to me like this would not work, since you equate "tree" with "cell_id" here. I think what you need to do instead is something like this (probably not functional code; also there's likely a better (faster) way to do this than searching `mesh.cell_ids` in each iteration):
```suggestion
    mesh_view_cell_id = 0
    for tree_id in eachindex(trees)
        tree_offset = trees[tree_id].quadrants_offset
        quadrants = unsafe_wrap_sc(p4est_quadrant_t, trees[tree_id].quadrants)

        for i in eachindex(quadrants)
            parent_cell_id = offset + i
            if !(parent_cell_id in mesh.cell_ids)
                # This cell is not part of the mesh view, thus skip it
                continue
            end
            
            mesh_view_cell_id += 1
            ...
            # `mesh_view_cell_id` is now what before was `element`
```